lib/puppet/provider/elb_loadbalancer/v2.rb

```diff
@@ -5,60 +5,114 @@
 
   mk_resource_methods
 
-  def self.instances
+  def self.instances(ref_catalog=nil)
+    Puppet.debug('Fetching ELB instances')
     regions.collect do |region|
-      retries = 0
-      begin
         load_balancers = []
-        region_client = elb_client(region)
-        region_client.describe_load_balancers.each do |response|
-          response.data.load_balancer_descriptions.collect do |lb|
-            load_balancers << new(load_balancer_to_hash(region, lb))
+        elbs do |lb|
+          retries = 0
+          begin
+            load_balancers << new(load_balancer_to_hash(region, lb, ref_catalog))
+          rescue Aws::EC2::Errors::RequestLimitExceeded => e
+            retries += 1
+            if retries <= 8
+              sleep_time = 2 ** retries
+              Puppet.debug("Request limit exceeded, retry in #{sleep_time} seconds")
+              sleep(sleep_time)
+              retry
+            else
+              raise PuppetX::Puppetlabs::FetchingAWSDataError.new(region, self.resource_type.name.to_s, e.message)
+            end
+          rescue Timeout::Error, StandardError => e
+            raise PuppetX::Puppetlabs::FetchingAWSDataError.new(region, self.resource_type.name.to_s, e.message)
           end
         end
+
         load_balancers
```

Comment:
This looks like it is indented two spaces too far? Maybe it's just github's diffing that is wrong.